### DISCUSSION TITLE
Failure with runAll flag testing

### DISCUSSION THREAD

**Discussion by Fracturist** [WENO-OF/WENOEXT#43] [Q&A] [answered]
Hi all,

I encountered a problem  when performing the test cases. After some minor modifications, all goes well with the tag "--runSimple". But following error appeared when I used the tag "--runAll"

```bash
/home/alex/OpenFOAM/alex-v2106/src/openSource/WENOEXT/tests/src/geometryWENO-BasicFunc-Test.C:305: FAILED:
  REQUIRE( Approx(mag(blaze::det(JInvI)))==1E+05 )
with expansion:
  Approx( 225000.0000000003 ) == 100000.0
```

I am not familiar with the CATCH2 framework.  Could someone help me fix this?

Best,

Fracturist

[logrunSimple.txt](https://github.com/WENO-OF/WENOEXT/files/8771713/logrunSimple.txt)
[logrunAll.txt](https://github.com/WENO-OF/WENOEXT/files/8771715/logrunAll.txt)

**Comment by JanGaertner** [comment, marked as answer]
This is not an error or bug, but a warning that one of the tests failed. Meaning, that something has changed to give now incorrect results. The expected determinant of the inverse Jacobi matrix for this test case is 1E+5 but your result is 225000, hence the test fails.

However, it is interesting that the test does not fail for the first run, because it should be executed for both.

I also see that you use the master branch version, please test the development branch version, which has improved features. I will update the master branch soon to fix this issue. 

Best,

Jan

**Comment by Fracturist** [comment]
Hi Jan,

Thank you very much for your explanation 😄. I am also comfused by the discrepancy between the two tests. From the runTest script, there should be no difference I think.

I will try the development branch version to see if it helps.

Best, 

Fracturist

**Comment by Fracturist** [comment]
Hi Jan,

I have tried the development branch and it works quite well for tests.

I work with v2106 and paraview 5.9.1 and have to make some slight changes to get the correct test report. I post them here in case someone might need these.

1. Change "baseCase" to "2DMesh-cyclic" of line 21 and 25 in testReport.html. I suppose there is a baseCase before?

https://github.com/WENO-OF/WENOEXT/blob/f152a37eff2512317a06a80368ed40efcaf0dc4a/tests/report/testReport.html#L20-L25

2. I think the original plotted lines are not correct so I changed to

```bash
plot "results.dat" every 2::0::2 using ($0+0.24):3 with boxes ls 2 notitle,\
     "results.dat" every ::3 using ($0+0.51):3 with boxes ls 3 title 'best cond.
```

https://github.com/WENO-OF/WENOEXT/blob/f152a37eff2512317a06a80368ed40efcaf0dc4a/tests/Cases/2DMesh-cyclic/PLOT/plotReport.gplt#L34-L35

Please correct me if I was wrong.

3. Change write interval  to 1 or there will be a set of empty time directories which will make [createFigures.py](https://github.com/WENO-OF/WENOEXT/blob/f152a37eff2512317a06a80368ed40efcaf0dc4a/tests/Cases/advectionCase/createFigures.py) fail. I think an alternative way is to modify [WENOUpwindFit-AdvectionTest.C](https://github.com/WENO-OF/WENOEXT/blob/f152a37eff2512317a06a80368ed40efcaf0dc4a/tests/src/WENOUpwindFit-AdvectionTest.C).

https://github.com/WENO-OF/WENOEXT/blob/f152a37eff2512317a06a80368ed40efcaf0dc4a/tests/Cases/advectionCase/system/controlDict#L31

 4. Change ``uncompressed`` to ``off`` for OF fire compression option if using recent version. This will make output cleaner. 
 
This is what i got in the end.

![Selection_002](https://user-images.githubusercontent.com/103815630/170479925-d7e4319c-3ecb-4dc1-97c4-70438f6c634d.png)

**Reply by JanGaertner** [reply]
Hello,

thank you for your investigation of the problem. If you want you can create a pull request to merge the changes of the tests into the development branch. I had forgotten about the test report this creates and had not updated the changes in paths etc. 

The results are as expected, so everything seems to be ok. The WENO result shows much less diffusion than the limited linear one after one rotation of the slotted disk. For the divergence test it is also as expected, as the error is reduced for the best conditioned setting and for the higher polynomial solution. 

Best,

Jan

**Comment by Fracturist** [comment]
Hi Jan,

Thank you for checking the results. I have created the pull request #44 to merge those changes.

Best,

Fracturist

**Reply by JanGaertner** [reply]
Hello Fracturist,

thank you, I have accepted the pull request #44.
May I ask for what use case you use WENO? I am intersted to see in what areas it is applied. 

Best,

Jan

**Reply by Fracturist** [reply]
Hi Jan,

I use LES to study transverse jets. I recently wanted to try some higher-order schemes. WENO is a good choice for me. 

By the way, I do want to thank you for your valuable work done so far. It really helped me save a lot of time.

Best,

Fracturist
